### PR TITLE
Article/frontend

### PR DESCRIPTION
**Pull request by krampstudio**
Frontend core documentation.
This is the first batch of the frontend documentation. Obviously it's not complete yet but it's a start and we will improve later.

This pull request contains content already reviewed (from @jsconan).

@dietertaotesting could you please verify the metadata, tagging and grouping (and the way to link articles together (like [here](https://github.com/oat-sa/taohub-articles/pull/48/files#diff-946e6a2f00e6b6b53c49ca5a6afb8f1cR394) and [there](https://github.com/oat-sa/taohub-articles/pull/48/files#diff-946e6a2f00e6b6b53c49ca5a6afb8f1cR109))
@spotz could you please quickly check my broken English
@jsconan could you please have a look on the content itself.

### CODE REVIEW

File: articles/Frontend/Architecture/amd-require.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
+<!--
+authors:
+    - "Bertrand Chevrier"
+tags:
+   - "Frontend":
+        - "Frontend Architecture"
+-->
```
[the remark this answers is not in the document]
Comment (krampstudio) [reply articles/Frontend/Architecture/amd-require.md]: Thank you for the input @dietertaotesting 
I was trying to achieve o have the following structure : 
 - Frontend
    -  Frontend Architecture
          - AMD and require.js
          - Build
          - ...
    -  Component
         - Date Time picker
         - Calculator
         - ...
    - SDK
       - Eventifier 
       - ...

But if we can have only one level of nesting, I'll change them

[the remark this answers is not in the document]
Comment (dietertaotesting) [reply articles/Frontend/Architecture/amd-require.md]: sorry, one level only right now :-(

File: articles/Frontend/Architecture/coding-guide.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+ - Code should be consistent and easy to understand and self descriptive.
+ - Writing comments when it is necessary is highly appreciated. Comments shouldn't be redundant with the code itself (clear code document itself) but provide additional and useful information.
+ - Variable, function, method names should reflect the intent in a clear way.
+ - Variable definitions should be separated always by new lines to help readability.
+ - Variable should be defined using it's own statement (no comma after the declaration)
+ - Use 4 spaces for indents
+ - No more than one blank line
+ - No ASCII art within the source code
+ - Use single quotes for string literals 
+ - Use template literals instead of concatenation with variables or for multi line strings
```
Comment (jsconan) [inline articles/Frontend/Architecture/coding-guide.md]: Is it for every flavour or only for ES6+ ?

File: articles/Frontend/Architecture/amd-require.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+# AMD & Require.js
+
+> How to deal with require.js and AMD into TAO.
+
+For historical reasons, JavaScript files in TAO are AMD modules.
```
Comment (spotz) [inline articles/Frontend/Architecture/amd-require.md]: For historical reasons, Javascript files are AMD modules in TAO.

File: articles/Frontend/Architecture/amd-require.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+TAO uses [AMD](https://en.wikipedia.org/wiki/Asynchronous_module_definition) and [require.js](https://requirejs.org/) :
+ - to define import and export, and declare dependencies
+ - as a dependency injection library
+ - as a dynamic loader
+ - dependency resolver (for bundling)
```
Comment (spotz) [inline articles/Frontend/Architecture/amd-require.md]: as a dependency resolver (for bundling)

File: articles/Frontend/Architecture/amd-require.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+## Module paths
+
+The dependencies are declared using a module name, which is not necessarily a path. By default you can make the reference to a module by using the following pattern:  `${extensionName}/${pathInViews}/${moduleName}`.
+For example to access `taoQtiTest/views/js/runner/plugins/control/timers/timer.js` you'll use `taoQtiTest/runner/plugins/control/timers/timer`.
+You'll notice the `views/js` disappear as well as the file extension.
+
+The base URL is always for JavaScript resources `tao/views/js` so for modules into the `tao` extension you don't need to prefix them with `tao`. For example to access `tao/views/js/core/eventifier.js`, `core/eventifier` should  be used.
+
+Modules with an alias defined in the configuration can be called using this alias `lodash`, `jquery`, `moment`, `i18n`, etc. Since alias create some coupling between the source code and the configuration, we try to reduce their usage as much as possible.
```
Comment (spotz) [inline articles/Frontend/Architecture/amd-require.md]: this defined alias such as

File: articles/Frontend/Architecture/amd-require.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+    'use strict';
+
+    return {
+        bar : true
+    }
+});
+```
+**They won't work once optimized**
+
+Some libraries still uses named modules, to support this heresy we have to define an alias in the configuration.
```
Comment (spotz) [inline articles/Frontend/Architecture/amd-require.md]: use vs uses

File: articles/Frontend/Architecture/amd-require.md [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+
+## Dependency injection
+
+The content of the callback function is executed this first time it is required by another module and the main scope lifecycle will remain until the page refresh.
```
Comment (spotz) [inline articles/Frontend/Architecture/amd-require.md]: the vs this

**Pull request merged**